### PR TITLE
Allow dynamic type generation in Arrays

### PR DESCRIPTION
**Pull request by inlinestyle**
Review @tleach,  feature implementation for issue https://github.com/gamechanger/schemer/issues/9

### CODE REVIEW

**Comment by inlinestyle** [comment]
@tleach Wasn't sure if this belonged in the `README` or not.

File: tests/schemer/sample.py [hunk 1]
```diff
@@ -66,7 +66,8 @@ def get_website_schema(document):
     "misc":             {"type": Mixed(basestring, int)},
     "linked_id":        {"type": Mixed(int, basestring)},
     "external_code":    {"type": basestring, "nullable": False},
-    "website":          {"type": get_website_schema}
+    "website":          {"type": get_website_schema},
+    "editors":          {"type": Array(lambda document: name_schema)}
```
Comment (tleach) [inline tests/schemer/sample.py]: Would be nice if this pointed to a separate function which returned a different type depending on some condition, to illustrate what this capability is actually good for.

Maybe have the `'editors'` list allow either a `str` for the name or a `name_schema`?

File: tests/schemer/schema_test.py [hunk 2]
```diff
@@ -278,6 +278,14 @@ def test_dynamic_function_wrong_type(self):
         self.document_1['website'] = 56
         self.assert_document_paths_invalid(self.document_1, ['website'])
 
+    def test_array_with_dynamic_function_correct_type(self):
+        self.document_1['editors'] = [{'first': 'Jordan', 'last': 'Gansey'}]
+        blog_post_schema.validate(self.document_1)
```
Comment (tleach) [inline tests/schemer/schema_test.py]: As hinted already, would be good to test the validation against multiple types.

**Comment by tleach** [comment]
Looks good. A slightly more stringent test and you're good to go.

**Comment by inlinestyle** [comment]
@tleach Updated the tests per your recommendations. I also added some assertions around heterogenous lists.

**Comment by tleach** [comment]
Looks good.

**Pull request merged**